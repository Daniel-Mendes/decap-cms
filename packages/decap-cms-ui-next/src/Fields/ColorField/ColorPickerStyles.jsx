import React from 'react';
import { Global, css, withTheme } from '@emotion/react';

function getColorPickerStyles(theme) {
  return css`
    .react-colorful .react-colorful__pointer {
      /* border: 2px solid ${theme.color.emphasis}; */
    }

    .react-colorful .react-colorful__pointer.react-colorful__saturation-pointer {
      width: 32px;
      height: 32px;
      border-radius: 8px;
    }

    .react-colorful .react-colorful__pointer.react-colorful__hue-pointer,
    .react-colorful .react-colorful__pointer.react-colorful__alpha-pointer {
      width: 16px;
      border-radius: 4px;
    }
  `;
}

function getDatetimePickerStyles({ theme }) {
  return <Global styles={getColorPickerStyles(theme)} />;
}

export default withTheme(getDatetimePickerStyles);
