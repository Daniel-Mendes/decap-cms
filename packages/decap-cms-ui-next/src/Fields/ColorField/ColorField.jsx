import React, { useState, useRef } from 'react';
import styled from '@emotion/styled';
import { HexColorPicker } from 'react-colorful';

import { Menu } from '../../Menu';
import TextField from '../TextField';
import ColorPickerStyles from './ColorPickerStyles';
import { useClickAway } from '../../hooks';

const StyledMenu = styled(Menu)`
  padding: 0;
`;

const Swatch = styled.div`
  position: absolute;
  right: ${({ inline }) => (inline ? 0 : 0.5)}rem;
  top: 1.85rem;

  width: 2rem;
  height: 2rem;
  border-radius: 6px;
  border: 3px solid ${({ theme }) => theme.color.surface};
  background-color: ${({ color }) => color};
  box-shadow: ${({ theme }) =>
    `0 0 0 1px ${theme.color.lowEmphasis}, inset 0 0 0 1px ${theme.color.lowEmphasis}`};
  cursor: pointer;
`;

function ColorField({
  name,
  label,
  status,
  placeholder,
  description,
  alpha,
  readOnly,
  value,
  onChange,
  error,
  errors,
  ...props
}) {
  const [color, setColor] = useState(value);
  const [anchorEl, setAnchorEl] = useState(null);
  const popover = useRef(null);

  useClickAway(popover, handleClose);

  function handleOpenMenu(event) {
    setAnchorEl(event.currentTarget);
  }

  function handleClose(event) {
    console.log('handleClose', event);
    // setAnchorEl(event ? event.currentTarget : null);
    setAnchorEl(null);
  }

  return (
    <>
      <ColorPickerStyles />

      <TextField
        name={name}
        label={label}
        labelTarget={name}
        status={status}
        placeholder={placeholder}
        description={description}
        onClick={readOnly ? handleOpenMenu : null}
        onChange={setColor}
        error={error}
        errors={errors}
        type={color}
        value={color}
        icon={<Swatch color={color} onClick={handleOpenMenu} />}
        readOnly={readOnly}
        {...props}
      >
        <StyledMenu
          anchorOrigin={{ x: 'right', y: 'bottom' }}
          transformOrigin={{ x: 'right', y: 'top' }}
          anchorEl={anchorEl}
          open={!!anchorEl}
          onClose={handleClose}
          ref={popover}
        >
          {anchorEl && (
            <HexColorPicker
              color={color}
              alpha={alpha}
              onChange={color => {
                setColor(color);
                onChange && onChange(color);
              }}
            />
          )}
        </StyledMenu>
      </TextField>
    </>
  );
}

export default ColorField;
