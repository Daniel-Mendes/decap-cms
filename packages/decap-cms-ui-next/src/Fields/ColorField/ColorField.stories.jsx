import React from 'react';
import styled from '@emotion/styled';

import ColorField from '.';

export default {
  title: 'Fields/ColorField',
  component: ColorField,
  argTypes: {
    errors: {
      control: 'object',
      if: {
        arg: 'error',
      },
    },
  },
  args: {
    label: 'Color Field Label',
    status: 'Required',
    placeholder: 'Type color here',
    description: 'This is a description',
    value: '#000000',
    readOnly: true,
    alpha: false,
    inline: false,
    error: false,
    errors: [
      {
        message: 'Error message.',
      },
    ],
  },
};

const StyledColorField = styled(ColorField)`
  width: 33vw;
`;

export function _ColorField(args) {
  return <StyledColorField {...args} />;
}
