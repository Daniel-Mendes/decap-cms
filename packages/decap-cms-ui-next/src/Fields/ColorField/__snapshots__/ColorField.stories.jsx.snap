// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fields/ColorField _ColorField smoke-test 1`] = `"<div class="eqka0b30 css-omvouz-FieldWrap-StyledColorField e1x0ipyd6"><div class="css-kuth3m-FieldInside-FieldInside-FieldInside-FieldInside e1x0ipyd5"><label class="css-1n9k11u-Label-StyledLabel e1x0ipyd3">Color Field Label <div color="neutral" radius="large" class="css-6930ng-BadgeWrap-BadgeWrap epfx4ki3">Required</div></label><div class="css-dj548v-ChildrenWrap e1x0ipyd4"><input readonly="" type="#000000" placeholder="Type color here" autocomplete="off" class="css-gfgrnx-StyledInput e4jt8f0" value="#000000"></div><div color="#000000" class="css-1o88qaz-Swatch e1eklj1j0"></div><p class="css-nalsj0-StyledDescription e1x0ipyd1">This is a description</p></div></div>"`;
